### DISCUSSION TITLE
Sort files by name using localeCompare

### DISCUSSION THREAD

**Discussion by uahnbu** [kimlimjustin/xplorer#196] [Ideas]
The default sort of Javascript will not work properly with files e.g. as chapters of a book.
Current sort behavior:
```
Chapter_1.txt
Chapter_11.txt
Chapter_2.txt
```
Expected behavior:
```
Chapter_1.txt
Chapter_2.txt
Chapter_11.txt
```
The natural sort can be achieved by using localeCompare, as in this [article](https://stackoverflow.com/questions/2802341/javascript-natural-sort-of-alphanumerical-strings#answer-38641281).
```javascript
var collator = new Intl.Collator(undefined, {numeric: true, sensitivity: 'base'});
var myArray = ['1_Document', '11_Document', '2_Document'];
console.log(myArray.sort(collator.compare));
```

**Comment by kimlimjustin** [comment]
Good pov, but I wonder how about sorting descendant using Intl collector, is there any way to do it natively? or does it have to be sorted using the method you provided and then reverse the array? 

Maybe this is a stupid question but I've never worked with Intl collator before, so yeah....

**Reply by uahnbu** [reply]
Actually
```javascript
new Intl.Collator(void 0, {numeric: true, sensitivity: 'base'}).compare
```
is the compare function.
You can simply return the reverse function of it.
```javascript
const comparator = new Intl.Collator(void 0, {numeric: true, sensitivity: 'base'}).compare;
return arr.sort((a, b) => comparator(b, a));

**Reply by kimlimjustin** [reply]
Ah okay, will implement it later this day
